### PR TITLE
Stdlib: add drop and dropWhile

### PR DESCRIPTION
**Pull request by tatchi**
Implement #310

### CODE REVIEW

File: stdlib/list.gr [hunk 1]
```diff
@@ -195,3 +195,25 @@ let unique = (list) => {
...
   iter(list, [])
 }
+
+let rec drop = (n, list) => {
+  if (n < 0) {
+    fail 'number of items to drop cannot be a negative number'
+  } else {
+    match ((n, list)) {
+      | (_,[]) => []
+      | (n, _) when n == 0 => list
```
Comment (tatchi) [inline stdlib/list.gr:205]: Looks like we can't pattern match on number so I used a guard here.

File: stdlib/list.gr [hunk 2]
```diff
@@ -195,3 +195,25 @@ let unique = (list) => {
...
+    }
+  }
+}
+
+let rec dropWhile = (fn, list) => {
+  match (list) {
+    | [] => list
+    | [first, ...rest] => if (fn(first)) dropWhile(fn, rest) else list
+  }
+}
```
Comment (tatchi) [inline stdlib/list.gr:216]: I'm not sure this one is tail-recursive but from what I saw, others from this file like `count` are not. I don't know at which point it matters?

File: stdlib/list.gr (hunk 1 above)
Comment (belph) [reply stdlib/list.gr:205]: @ospencer Is this a bug or a TODO?

File: stdlib/list.gr (hunk 2 above)
Comment (belph) [reply stdlib/list.gr:216]: Hmm, I think we want to move towards making both this and `count` tail-recursive. @ospencer should we make a separate issue for this, or do we want to include making this/these tail-recursive inside of this PR?

**Review by belph — COMMENTED** [review]
This is great stuff, thank you! I've requested a review from @ospencer so that we can address the comments you've left, but I have no issues merging this once those are ironed out.

File: stdlib/list.gr (hunk 1 above)
Comment (ospencer) [reply stdlib/list.gr:205]: We already have an issue: https://github.com/grain-lang/grain/issues/284

File: stdlib/list.gr (hunk 2 above)
Comment (ospencer) [reply stdlib/list.gr:216]: Maybe it's really early for me, but is this not tail-recursive? The call to `dropWhile` is in the tail position of the true branch of a tail position `if`.

`count` does need to be changed though. Made https://github.com/grain-lang/grain/issues/319.

**Review by ospencer — APPROVED** [review]
Thanks @tatchi!

File: stdlib/list.gr (hunk 2 above)
Comment (belph) [reply stdlib/list.gr:216]: Oh, yes, lol. I missed that; you're right.

Comment (tatchi) [reply stdlib/list.gr:216]: Thanks for the review, I was indeed not sure for the `dropWhile` :)

**Pull request merged**